### PR TITLE
Update list view item height programmatically

### PR DESCRIPTION
**Pull request by rebornix**
This PR contains two changes to the List View as we discussed today @joaomoreno 

* `updateElementHeight` which can modify the size of a list item and rerender the view properly
* `triggerScrollFromMouseWheelEvent ` which can accept mouse wheel events from external sources and scroll accordingly

Like all other rendering methods in the List View, if the List View is in the middle of rendering, invoking another `render()` or `updateElementHeight()` might mess up the view. This is a convention shared by all List View users.

### CODE REVIEW

**Comment by rebornix** [comment]
This PR will be merged after @joaomoreno reviews and only after the master is open for business.

File: src/vs/base/browser/ui/list/listView.ts [hunk 1]
```diff
@@ -273,6 +275,28 @@ export class ListView<T> implements ISpliceable<T>, IDisposable {
...
+	triggerScrollFromMouseWheelEvent(browserEvent: IMouseWheelEvent) {
+		this.scrollableElement.triggerScrollFromMouseWheelEvent(browserEvent);
+	}
+
+	updateElementHeight(index: number, element: T, size: number): void {
+		const lastRenderRange = this.getRenderRange(this.lastRenderTop, this.lastRenderHeight);
+
+		let heightDiff = index < lastRenderRange.start ? size - this.items[index].size : 0;
+		this.rangeMap.splice(index, 1, [{ size: size }]);
+
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/list/listView.ts]:
```suggestion
```

**Review by joaomoreno — APPROVED** [review]
All good, just small changes suggested.

**Comment by rebornix** [comment]
Love the code suggestion feature from GitHub ;)

**Pull request merged**